src/init.cpp

```diff
@@ -1504,9 +1510,13 @@ bool AppInitMain(NodeContext& node, interfaces::BlockAndHeaderTipInfo* tip_info)
     LogPrintf("* Using %.1f MiB for in-memory UTXO set (plus up to %.1f MiB of unused mempool space)\n", cache_sizes.coins * (1.0 / 1024 / 1024), mempool_opts.max_size_bytes * (1.0 / 1024 / 1024));
 
     for (bool fLoaded = false; !fLoaded && !ShutdownRequested(node);) {
-        node.mempool = std::make_unique<CTxMemPool>(mempool_opts);
+        node.mempool = std::make_unique<CTxMemPool>(mempool_opts, validation_signals);
```

Comment:
> It might also be worth asking whether the signals parameter actually should be required.

It is used in `CTxMemPool` methods, so should be required, no? Or are you saying these classes should be given the option to not be sending signals in the first place? In that case, I don't see what the difference between the mempool and the other validation classes is. The signals could be optional in the chainman too.

Reply:
re: https://github.com/bitcoin/bitcoin/pull/28960#discussion_r1458838935

> The signals could be optional in the chainman too.

Yes, absolutely. The question is in cases where libbitcoinkernel applications don't care about receiving signals (if maybe just they are just validating a block), should they be forced to declare `ValidationSignals validation_signals{std::make_unique<util::ImmediateTaskRunner>()};` and pass a `validation_signals` parameter to the ChainstateManager object that won't do any real work. Eliminating this requirement might be good for libbitocoinkernel usability, but it would make validation code a little more verbose, because it would need to check if the signals object is null before sending signal. I don't have a real opinion about this, so I think the current approach seems pretty reasonable.

The mempool case is slightly different though. It sends fewer signals than validation code, and maybe requiring a new CTxMempool constructor argument will add pointless verbosity to mempool tests. So I think the mempool case is more worth thinking about and maybe getting some feedback from others familiar the mempool code. They may prefer to let this be optional.